src/keboola_mcp_server/tools/components/validation.py

```diff
@@ -0,0 +1,145 @@
+import logging
+from typing import Any, Dict, List, Optional, Tuple, Annotated
+
+from jsonschema import Draft7Validator, exceptions
+from mcp.server.fastmcp import Context
+from pydantic import Field
+
+from keboola_mcp_server.client import KeboolaClient
+from keboola_mcp_server.tools.components.utils import _get_component_details
+
+LOG = logging.getLogger(__name__)
+
+
+async def fetch_component_schema(
+        client: KeboolaClient, component_id: str
+) -> Tuple[Optional[Dict[str, Any]], Optional[Dict[str, Any]]]:
+    """
+    Fetch configuration schemas directly from storage API for a specific component.
+
+    Args:
+        client: The Keboola client
+        component_id: The ID of the component
+
+    Returns:
+        A tuple of (root_configuration_schema, row_configuration_schema)
+    """
+    try:
+        endpoint = 'v2/storage/components'
```

Comment:
This will only give you components in a project. There is no reason imo to have another call to storage API here. The root and row schema is available in the component detail that uses the index in the AI service